src/WebJobs.Extensions.DurableTask.Analyzers/SyntaxNodeUtils.cs

```diff
@@ -47,137 +48,138 @@ public static bool IsInsideOrchestrator(SyntaxNode node)
             return false;
         }
 
-        internal static bool TryGetClassSymbol(SyntaxNode node, SemanticModel semanticModel, out INamedTypeSymbol classSymbol)
+        public static bool IsMarkedDeterministic(SyntaxNode node) => TryGetDeterministicAttribute(node, out _);
+
+
+        public static bool TryGetDeterministicAttribute(SyntaxNode node, out SyntaxNode deterministicAttribute) => TryGetAttribute(node, "Deterministic", out deterministicAttribute);
```

Comment:
This can probably be marked private, because I don't see us ever needing to get the actual `DeterministicAttribute`, we just want to know if it is present.

Reply:
It's used in the CodeFixProviders actually. I noticed some of my methods are public and some are internal in SyntaxNodeUtils though. I should probably make them all the same (unless they're private).